### ISSUE TITLE
TrainingStateController threshold implementations are incorrect

### ISSUE THREAD

**Issue by sdrobert** [sdrobert/pydrobert-pytorch#55]
Checking for early stopping and reducing learning rates in TrainingStateController compare each epoch's metric to the best score, *not* the score which we last reset the patience at. This causes a slippery slope: if my threshold is 1 and I reduce each epoch by only 0.5, I'll end up tripping up the mechanism (since the last epoch becomes the new best each time).

**Comment by sdrobert** [comment]
Fixed in #56
